packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -534,6 +535,12 @@ class SemanticsData with Diagnosticable {
   /// The reading direction is given by [textDirection].
   final String tooltip;
 
+  /// If non-null, indicates that this subtree represents a heading.
+  ///
+  /// The value should be a number between 1 and 6, indicating the hierarchical
+  /// level as a heading.
+  final int? headingLevel;
```

Comment:
same here, we should not use null in this class, as there isn't a way to represent null in engine

Reply:
Done, I'm initially making the attribute non required because otherwise there will be a lot of failures in integration tests. We could make it required later, I hope that works for you :)